### ISSUE TITLE
Unable to upload Solang generated idl

### ISSUE THREAD

**Issue by CrossedCarpet** [solana-foundation/anchor#2679]
Steps to reproduce:

- `anchor init testprogram --solidity`
- `anchor build`
- `anchor keys sync`
- ``` anchor idl init -f target/idl/testprogram.json `solana address -k target/deploy/testprogram-keypair.json` ```

This last commands outputs:
```sh
Error: RPC response error -32002: Transaction simulation failed: Error processing Instruction 0: invalid program argument [3 log messages]

Caused by:
    RPC response error -32002: Transaction simulation failed: Error processing Instruction 0: invalid program argument [3 log messages]
```
I haven't been able to find any solution for this. I also do not know how to access those `[3 log messages]`. 

edit
---  
These are my program versions in case it's some compatibility issue:  
- solana-cli 1.16.17  
- anchor-cli 0.28.0  
- solang version v0.3.2

**Comment by acheroncrypto** [comment]
@seanyoung any insight?

**Comment by seanyoung** [comment]
I'm aware of this, I have not fully debugged it yet. It certainly looks like a solang issue.

**Comment by CrossedCarpet** [comment]
Is there a workaround where I can load the idl from a file within my webapp?

**Comment by acheroncrypto** [comment]
> Is there a workaround where I can load the idl from a file within my webapp?

Yes, the IDL is just a JSON file, you can load it from anywhere and pass it to `new anchor.Program(idl, ...)`.

**Comment by CrossedCarpet** [comment]
Ah that's right, thanks!

I was looking for a way to modify: `const idl = await Program.fetchIdl(programID, getProvider());`

But a simple `import idl from "./testprogram.json";` suffices.

**Comment by conky-dev** [comment]
I am having issue initing my idl too. Worth noting that I am able to upgrade existing idls. But unable to init a new idl.

**Comment by acheroncrypto** [comment]
@conky-dev @amilkov3

This issue is related to Solang only, please create a new issue if you don't use Solang.

**Comment by seanyoung** [comment]
I've been working on this issue the last couple of days. I should have an update soon.
